### PR TITLE
better publish errors

### PR DESCRIPTION
**Pull request by jackrobison**

### CODE REVIEW

File: lbrynet/daemon/Daemon.py [hunk 1]
```diff
@@ -1841,8 +1842,25 @@ def jsonrpc_publish(self, name, bid, metadata=None, file_path=None, fee=None, ti
...
+            if not os.path.isfile(file_path):
+                raise Exception("invalid file path to publish")
+            claim_copy['stream']['source'] = {
+                'version': '_0_0_1',
+                'sourceType': 'lbry_sd_hash',
+                'source': '0' * 96,
+                'contentType': ''
+            }
+        else:
+            raise Exception("no source provided to publish")
```
Comment (mirgee) [inline lbrynet/daemon/Daemon.py]: The documentation says "If no path is given but a metadata dict is the source from the given metadata will be used". This doesn't seem to be intended behavior here. Maybe update the doc, or change the logic?

**Comment by mirgee** [comment]
Shouldn't this be linked to #901?

**Comment by lyoshenka** [comment]
please rebase and merge @jackrobison

File: lbrynet/daemon/Daemon.py (hunk 1 above)
Comment (jackrobison) [reply lbrynet/daemon/Daemon.py]: updated the docstring

**Pull request merged**